src/Markdig/Helpers/LinkHelper.cs

```diff
@@ -690,7 +690,7 @@ public static bool IsValidDomain(string link, int prefixLength)
 
                 if (!c.IsAlphaNumeric())
                 {
-                    if (c == '/') // End of domain name
+                    if (c == '/' || c == '?' || c == '#' || c == ':') // End of domain name

```

Comment:
Just an opinion:

this could perhaps also be written as:
```cs
if ("/?#:".Contains(c))
```

This would allow it to be extended to more characters in the future if needed.
Don't know if there is any performance difference though.

Reply:
Up to personal taste I guess.
Performance-wise it doesn't really matter in this case, I just try to stick to the same style as the rest of the library.

Reply:
Just made a benchmark comparison for 5 characters in a tests (between contains and ==)

```
   Method |     Mean |     Error |    StdDev |   Median |
--------- |---------:|----------:|----------:|---------:|
 Contains | 641.7 ns | 12.833 ns | 31.959 ns | 629.5 ns |
   Equals | 254.1 ns |  5.030 ns |  8.403 ns | 253.0 ns |
```
And it seems that Equals is still faster. So yeah, let's keep equals in that case